### DISCUSSION TITLE
tutorials/dataloaders-dgs/04-registering-a-dataloader

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#145] [course feedback]
# tutorials/dataloaders-dgs/04-registering-a-dataloader

Creating a data loader class

https://www.apollographql.com/tutorials/dataloaders-dgs/04-registering-a-dataloader

**Comment by andywgarcia** [comment]
There are more BatchLoader interfaces that can be implemented for the data loader such as BatchLoaderWithContext which allows for accessing the context within the load function. 

I think it would also be good to link to the DGS docs through the tutorial so we can get further information on each piece of DGS being used.

**Reply by lizhennessy** [reply]
Thanks for your feedback @andywgarcia ! We've made some updates so as not to sound so restrictive regarding the batch loading interface one chooses to use. Do let us know if you see anything else that you feel could use some improvement or polish!
